### PR TITLE
gateway-gcs: use minio.sys.temp/multipart/v1 as url base

### PR DESCRIPTION
**Pull request by krishnasrinivas**
## Description

This helps when we want to use a different backend format in future.

As discussed in today's meeting we will handle the backend format this way. i.e when we move to a different format we will start using minio.sys.temp/multipart/v2 and not worry about migrating the data in v1

## Motivation and Context

To support change in backend format for multipart files.

## How Has This Been Tested?

manual

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/gateway-gcs.go [hunk 1]
```diff
@@ -43,7 +43,9 @@ const (
 	// ListObjects we filter out this entry.
 	gcsMinioPath = "minio.sys.temp"
 	// Path where multipart objects are saved.
-	gcsMinioMultipartPath = gcsMinioPath + "/multipart"
+	// If we change the backend format we will use a different url path like /multipart/v2
+	// but we will not migrate old data.
+	gcsMinioMultipartPath = gcsMinioPath + "/multipart/v1"
```
Comment (harshavardhana) [inline cmd/gateway-gcs.go]: Perhaps call this `gcsMinioMultipartPathV1` ?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/gateway-gcs.go (hunk 1 above)
Comment (krishnasrinivas) [reply cmd/gateway-gcs.go]: will do that ... to be consistent with the rest of the code base

**Review by harshavardhana — APPROVED** [review]
LGTM and works.

**Review by balamurugana — APPROVED** [review]
Not tested

**Pull request merged**